test/e2e/cassandracluster_test.go

```diff
@@ -277,6 +279,159 @@ func cassandraClusterUpdateConfigMapTest(t *testing.T, f *framework.Framework, c
+	if err != nil {
+		t.Fatalf("Failed to list pods: %s", err)
+	}
+	assert.Equal(t, 2, len(pods.Items))
+
+	for _, pod := range pods.Items {
+		pod.Labels["operation-name"] = "cleanup"
+		pod.Labels["operation-status"] = "ToDo"
+		if err := f.Client.Update(goctx.TODO(), &pod); err != nil {
+			t.Fatalf("pod update failed: %s", err)
+		}
+	}
+
+	checkCleanupExecuted := func(rack string, node int, cc *api.CassandraCluster) (bool, error) {
+		dcRack := "dc1-" + rack
+		nodeName := fmt.Sprintf("%s-%s-%d", clusterName, dcRack, node)
+
+		//dcRackStatus, found :=cc.Status.CassandraRackStatus[dcRack]
+		_, found :=cc.Status.CassandraRackStatus[dcRack]
+		if !found {
+			return false, fmt.Errorf("Did not find rack status for %s", rack)
+		}
+
+		// Commenting out status checks for now due to https://github.com/Orange-OpenSource/cassandra-k8s-operator/issues/98
+		//
+		//if len(dcRackStatus.PodLastOperation.PodsOK) == 0 {
+		//	// The operation has not completed yet
+		//	return false, fmt.Errorf("The cleanup operation has not yet finished for %s", nodeName)
+		//}
+		//
+		//if len(dcRackStatus.PodLastOperation.PodsOK) > 1 {
+		//	// We only scheduled cleanup on one C* node in each rack, so PodsOK should have a length of 1.
+		//	return false, fmt.Errorf("expected cleanup to run on one pod it ran on %d. dcRackStatus.PodLastOperation (+%v)",
+		//		len(dcRackStatus.PodLastOperation.PodsOK), dcRackStatus.PodLastOperation)
+		//}
+		//if dcRackStatus.PodLastOperation.PodsOK[0] != nodeName {
+		//	// Make sure the operation executed against the expected node
+		//	return false, fmt.Errorf("expected cleanup to run on %s but it ran on %s", nodeName,
+		//		dcRackStatus.PodLastOperation.PodsOK[0])
+		//}
+
+		pod, err := f.KubeClient.CoreV1().Pods(namespace).Get(nodeName, metav1.GetOptions{})
+		if err != nil {
+			t.Logf("Failed to get pod %s: %s", nodeName, err)
+			return false, nil
+		}
+
+		val, exists := pod.Labels["operation-name"]
+		if !exists {
+			t.Logf("Expected to find label operation-name on %s", nodeName)
+			return false, nil
+		}
+		if val != "cleanup" {
+			t.Logf("Expected to find label operation-name=cleanup on %s but found operation-name=%s",
+				nodeName, val)
+			return false, nil
+		}
+
+		val, exists = pod.Labels["operation-status"]
+		if !exists {
+			t.Logf("Expected to find label operation-status on %s", nodeName)
+			return false, nil
+		}
+		if val != "Done" {
+			t.Logf("Expected to find label operation-status=Done on %s but found operation-status=%s",
+				nodeName, val)
+			return false, nil
+		}
+
+		val, exists = pod.Labels["operation-start"]
+		if !exists {
+			t.Logf("Expected to find label operation-start on %s", nodeName)
+			return false, nil
+		}
+
+
+		val, exists = pod.Labels["operation-end"]
+		if !exists {
+			t.Logf("Expected to find label operation-end on %s", nodeName)
+			return false, nil
+		}
+
+		// TODO parse start/end times and verify end > start
```

Comment:
Seems there is still work to do ? ^

Reply:
@cscetbon I am not sure how to parse the timestamps. Initially I thought https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go was used but learned that is not the case. Can you point me to an example of how to do the parsing?

Reply:
@jsanda easy, just use https://github.com/jsanda/cassandra-k8s-operator/blob/cleanup-e2e/pkg/k8s/util.go#L177 and then you can compare dates. Have a look also at https://github.com/jsanda/cassandra-k8s-operator/blob/cleanup-e2e/pkg/k8s/util_test.go#L36-L42

Reply:
Thanks!